app/src/pages/inside/dashboardItemPage/modals/widgetWizardModal/widgetWizardContent/wizardControlsSection/wizardThirdStepForm/wizardThirdStepForm.jsx

```diff
@@ -69,23 +94,51 @@ export class WizardThirdStepForm extends Component {
     return `${generatedName}_${this.getUniqPostfix()}`;
   };
 
-  initializeControlsForm = (
-    data = {
-      name: this.generateWidgetName(),
+  initializeControlsForm = () => {
+    const { dashboards, intl } = this.props;
+    const newDashboard = {
+      name: intl.formatMessage(messages.newDashboardName),
       description: '',
       share: false,
-    },
-  ) => this.props.initializeWizardThirdStepForm(data);
-
+    };
+    const selectedDashboard = dashboards.length > 0 ? dashboards[0] : newDashboard;
+    const data = {
+      name: this.generateWidgetName(),
+      description: '',
+      share: selectedDashboard.share,
+      selectedDashboard,
+    };
+    return this.props.initializeWizardThirdStepForm(data);
+  };
+  handleChange = (event, value) => {
+    const { formValues, change } = this.props;
+    const share = value.share ? value.share : formValues.share;
```

Comment:
I think that `share` field value should be by default as on selected dashboard. So when I choose not shared dashboard, value should be changed to `false` (as on 4th version).

Or it is an improvement? (in this case better `value.share || formValues.share`)